File: in_Java/0693 Binary Number with Alternating Bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        int cur = n & 1;
        n >>=1;
        while(n != 0){
            int newcur = n & 1;
            if(newcur == cur) return false;
            cur = newcur;
            n /= 2;
        }
        return true;
    }
}

File: in_Python/0693 Binary Number with Alternating Bits.py
class Solution(object):
    def hasAlternatingBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        pre = n % 2
        n /= 2
        while n != 0:
            cur = n % 2
            if pre == cur:
                return False
            n /= 2
            pre = cur
        return True